### PR TITLE
Service Registry Discussion

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
To investigate:
- akka RoutingActor
- Forward messages?

**Comment by cjllanwarne** [comment]
This should be rebased onto develop following #677

File: backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala [hunk 1]
```diff
@@ -15,6 +15,9 @@ trait WorkflowBackendActorFactory {
     */
   //TODO: Add backend configuration arg.
   //TODO: discuss about actorSystem... should be a backend actor system or the same from engine.
-  def getBackend(initClass: String, actorSystem: ActorSystem, workflowDescriptor: WorkflowDescriptor): ActorRef
+  def getBackend( initClass: String,
+                  actorSystem: ActorSystem,
+                  workflowDescriptor: WorkflowDescriptor,
+                  serviceRegistryActor: ActorRef): ActorRef
 
 }
```
Comment (mcovarr) [inline backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala]: \n

**Comment by scottfrazer** [comment]
I'm going to close this because the rebase was painful... I created a new branch `service_registry` and copied the relevant code in

**Pull request closed without merging**